test/js/web/fetch/fetch.test.ts

```diff
@@ -1816,3 +1816,84 @@ it("304 not modified with 0 content-length does not cause a request timeout", as
   expect(await response.arrayBuffer()).toHaveLength(0);
   server.stop(true);
 });
+
+describe("http/1.1 response body length", () => {
+  // issue #6932 (support response without Content-Length and Transfer-Encoding) + some regression tests
+
+  let server: net.Server | undefined;
+  beforeAll(async () => {
+    server = net.createServer(socket => {
+      socket.on("data", data => {
+        const lines = data.toString().split("\r\n");
```

Comment:
This for-loop is confusing to read, since the logic will essentially terminate on the first iteration. Could we change it to something like:

```js
const text = data.toString();
if (text.startsWith("GET /text") {
  // ...
} else if (text.startsWith("GET /json") {
  // ...
}